crates/interpreter/src/interpreter.rs

```diff
@@ -169,8 +146,10 @@ impl Interpreter {
 
     /// Returns a copy of the interpreter's return value, if any.
     #[inline]
-    pub fn return_value(&self) -> Bytes {
-        self.return_value_slice().to_vec().into()
+    pub fn return_value(&mut self) -> Bytes {
+        let return_data = self.return_value_slice().to_vec().into();
+        self.shared_memory.free_context_memory();
```

Comment:
I don't think this is great, can we keep this as is (to return a copy), and move the free to a finish/reset function (maybe it should also consume `self` instead of taking `&mut self`?)

Reply:
I don't think that consuming `self` is really that useful: this function is called inside `run_interpreter` but there we want to (probably in this order):
- save and return the return_data
- save return the gas used
- free the context memory
I cannot free the context memory after calling `return_value` is it consumes `self`.
However, I moved the free call just before the end of the `run_interpreter` call and I think it makes sense: in the same scope you create and free the context memory.
Let me know if it ok now